src/Installer/redist-installer/targets/RestoreLayout.targets

```diff
@@ -0,0 +1,331 @@
+      <BaseUrl>$(NetRuntimeRootUrl)</BaseUrl>
+      <DownloadFileName>$(DownloadedHostFxrInstallerFileName)</DownloadFileName>
+    </BundledInstallerComponent>
+
+    <BundledInstallerComponent Include="DownloadedNetCoreAppTargetingPackInstallerFile"
+                                Condition="'$(InstallerExtension)' != '' And (!$(Architecture.StartsWith('arm')) or '$(Rid)' == 'win-arm64' or '$(Rid)' == 'osx-arm64')">
+      <BaseUrl>$(NetRuntimeRootUrl)</BaseUrl>
+      <DownloadFileName>$(DownloadedNetCoreAppTargetingPackInstallerFileName)</DownloadFileName>
+    </BundledInstallerComponent>
+
+    <!-- TODO: Should we somehow obtain a .NET Standard ARM64 package? -->
+    <BundledInstallerComponent Include="DownloadedNetStandardTargetingPackInstallerFile"
+                          Condition="'$(InstallerExtension)' != '' And (!$(Architecture.StartsWith('arm')) or '$(Rid)' == 'osx-arm64')">
+      <BaseUrl>$(NetStandardTargetingPackRootUrl)</BaseUrl>
+      <DownloadFileName>$(DownloadedNetStandardTargetingPackInstallerFileName)</DownloadFileName>
+    </BundledInstallerComponent>
+
+    <BundledInstallerComponent Include="DownloadedNetCoreAppHostPackInstallerFile"
+                    Condition="'$(InstallerExtension)' != '' And (!$(Architecture.StartsWith('arm')) or '$(Rid)' == 'win-arm64' or '$(Rid)' == 'osx-arm64')">
+      <BaseUrl>$(NetRuntimeRootUrl)</BaseUrl>
+      <DownloadFileName>$(DownloadedNetCoreAppHostPackInstallerFileName)</DownloadFileName>
+    </BundledInstallerComponent>
+
+    <!-- These are apphosts from a different vertical which require a join. They are part of
+         the windows bundle which makes it also require a join. Disable for now in the VMR.
+         https://github.com/dotnet/source-build/issues/4777. -->
+    <BundledInstallerComponent Include="DownloadedAlternateNetCoreAppHostPackInstallerFile"
+                                Condition="'$(InstallerExtension)' == '.msi' and !$(Architecture.StartsWith('arm')) and '$(DotNetBuild)' != 'true'">
+      <BaseUrl>$(NetRuntimeRootUrl)</BaseUrl>
+      <DownloadFileName>$(DownloadedAlternateNetCoreAppHostPackInstallerFileName)</DownloadFileName>
+    </BundledInstallerComponent>
+
+    <!-- These are apphosts from a different vertical which require a join. They are part of
+         the windows bundle which makes it also require a join. Disable for now in the VMR.
+         https://github.com/dotnet/source-build/issues/4777. -->
+    <BundledInstallerComponent Include="DownloadedArm64NetCoreAppHostPackInstallerFile"
+                                Condition="'$(InstallerExtension)' == '.msi' And !$(Architecture.StartsWith('arm'))  and '$(DotNetBuild)' != 'true'">
+      <BaseUrl>$(NetRuntimeRootUrl)</BaseUrl>
+      <DownloadFileName>$(DownloadedArm64NetCoreAppHostPackInstallerFileName)</DownloadFileName>
+    </BundledInstallerComponent>
```

Comment:
It looks like the cross-arch app hosts are still disabled. Can we enable these now?

Reply:
I'm surprised that the build didn't fail without the packages restored. Thanks for catching this. This should now be fixed.